webpack/JobInvocationDetail/JobInvocationHostTable.js

```diff
@@ -0,0 +1,210 @@
+import TableIndexPage from 'foremanReact/components/PF4/TableIndexPage/TableIndexPage';
+import { useSetParamsAndApiAndSearch } from 'foremanReact/components/PF4/TableIndexPage/Table/TableIndexHooks';
+import {
+  useBulkSelect,
+  useUrlParams,
+} from 'foremanReact/components/PF4/TableIndexPage/Table/TableHooks';
+import Pagination from 'foremanReact/components/Pagination';
+import { getControllerSearchProps } from 'foremanReact/constants';
+import Columns, {
+  JOB_INVOCATION_HOSTS,
+  STATUS_UPPERCASE,
+} from './JobInvocationConstants';
+
+const JobInvocationHostTable = ({ id, targeting, finished, autoRefresh }) => {
+  const columns = Columns();
+  const columnNamesKeys = Object.keys(columns);
+  const apiOptions = { key: JOB_INVOCATION_HOSTS };
+  const {
+    searchParam: urlSearchQuery = '',
+    page: urlPage,
+    per_page: urlPerPage,
+  } = useUrlParams();
+  const defaultParams = { search: urlSearchQuery };
+  if (urlPage) defaultParams.page = Number(urlPage);
+  if (urlPerPage) defaultParams.per_page = Number(urlPerPage);
+  const { response, status, setAPIOptions } = useAPI(
+    'get',
+    `/api/job_invocations/${id}/hosts`,
+    {
+      params: { ...defaultParams, key: JOB_INVOCATION_HOSTS },
+    }
+  );
+
+  const combinedResponse = {
+    response: {
+      search: urlSearchQuery,
+      can_create: false,
+      results: response?.results || [],
+      total: response?.total || 0,
+      per_page: response?.perPage,
+      page: response?.page,
+      subtotal: response?.subtotal || 0,
+      message: response?.message || 'error',
+    },
+    status,
+    setAPIOptions,
+  };
+
+  const { setParamsAndAPI, params } = useSetParamsAndApiAndSearch({
+    defaultParams,
+    apiOptions,
+    setAPIOptions: combinedResponse.setAPIOptions,
+  });
+
+  const { updateSearchQuery } = useBulkSelect({
+    initialSearchQuery: urlSearchQuery,
+  });
+
+  const controller = 'hosts';
+  const memoDefaultSearchProps = useMemo(
+    () => getControllerSearchProps(controller),
+    [controller]
+  );
+  memoDefaultSearchProps.autocomplete.url = foremanUrl(
+    `/${controller}/auto_complete_search`
+  );
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      if (!finished || autoRefresh) {
+        setAPIOptions(prevOptions => ({
+          ...prevOptions,
+          params: {
+            ...prevOptions.params,
+          },
+        }));
+      }
+    }, 5000);
+
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, [finished, autoRefresh, setAPIOptions]);
+
+  const onPagination = newPagination => {
+    setParamsAndAPI({
+      ...params,
+      ...newPagination,
+      search: urlSearchQuery,
+    });
+  };
+
+  const bottomPagination = (
+    <Pagination
+      ouiaId="table-hosts-bottom-pagination"
+      key="table-bottom-pagination"
+      page={params.page}
+      perPage={params.perPage}
+      itemCount={response?.subtotal}
+      onChange={onPagination}
+    />
+  );
```

Comment:
This is the same code as the table pagination, since this pr is merged already can you remove it in a later  pr?